scripts/bert/create_pretraining_data.py

```diff
@@ -122,84 +155,120 @@ def __str__(self):
     def __repr__(self):
         return self.__str__()
 
-
-def write_instance_to_example_files(instances, tokenizer, max_seq_length,
-                                    max_predictions_per_seq, output_files):
-    """Create RecordIO files from `TrainingInstance`s."""
-    writers = []
-    for output_file in output_files:
-        writers.append(
-            mx.recordio.MXIndexedRecordIO(
-                os.path.splitext(output_file)[0] + '.idx', output_file, 'w'))
-
-    writer_index = 0
-
-    total_written = 0
+def transform(instance, tokenizer, max_seq_length, max_predictions_per_seq, do_pad=True):
+    """Transform instance to inputs for MLM and NSP."""
     pad = tokenizer.convert_tokens_to_ids(['[PAD]'])[0]
-    for (inst_index, instance) in enumerate(instances):
-        input_ids = tokenizer.convert_tokens_to_ids(instance.tokens)
-        input_mask = [1] * len(input_ids)
-        segment_ids = list(instance.segment_ids)
-        assert len(input_ids) <= max_seq_length
-
+    input_ids = tokenizer.convert_tokens_to_ids(instance.tokens)
+    input_mask = [1] * len(input_ids)
+    segment_ids = list(instance.segment_ids)
+    assert len(input_ids) <= max_seq_length
+    valid_lengths = len(input_ids)
+
+    masked_lm_positions = list(instance.masked_lm_positions)
+    masked_lm_ids = tokenizer.convert_tokens_to_ids(
+        instance.masked_lm_labels)
+    masked_lm_weights = [1.0] * len(masked_lm_ids)
+    masked_lm_valid_lengths = len(masked_lm_ids)
+    if do_pad:
         while len(input_ids) < max_seq_length:
             input_ids.append(pad)
             # Padding index MUST be defined to 0 on input_mask, segment_ids
             input_mask.append(0)
             segment_ids.append(0)
+        while len(masked_lm_positions) < max_predictions_per_seq:
+            masked_lm_positions.append(0)
+            masked_lm_ids.append(pad)
+            masked_lm_weights.append(0.0)
 
         assert len(input_ids) == max_seq_length
         assert len(input_mask) == max_seq_length
         assert len(segment_ids) == max_seq_length
 
-        masked_lm_positions = list(instance.masked_lm_positions)
-        masked_lm_ids = tokenizer.convert_tokens_to_ids(
-            instance.masked_lm_labels)
-        masked_lm_weights = [1.0] * len(masked_lm_ids)
-
-        while len(masked_lm_positions) < max_predictions_per_seq:
-            masked_lm_positions.append(0)
-            masked_lm_ids.append(0)
-            masked_lm_weights.append(0.0)
+    next_sentence_label = 1 if instance.is_random_next else 0
+
+    features = collections.OrderedDict()
```

Comment:
what's the purpose of using OrderedDict? I can't seem to find an obvious place where order matters.